pygmt/src/subplot.py

```diff
@@ -0,0 +1,222 @@
+        same for the entire subplot. Append **+p** to make all annotations
+        axis-parallel [horizontal]; if not used you may have to set
+        **clearance** to secure extra space for long horizontal annotations.
+
+        Notes for sharex/sharey:
+
+        - Labels and titles that depends on which row or column are specified
+          as usual via a subplot's own **frame** setting.
+        - Append **+w** to the **figsize** or **subsize** argument to draw
+          horizontal and vertical lines between interior panels using selected
+          pen [no lines].
+    title : str
+        While individual subplots can have titles (see **layout** or
+        **frame**), the entire figure may also have an overarching *heading*
+        [no heading]. Font is determined by setting :gmt-term:`FONT_HEADING`.
+    {V}
+    {XY}
+    """
+    kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+    # allow for spaces in string with needing double quotes
+    kwargs["A"] = f'"{kwargs.get("A")}"' if kwargs.get("A") is not None else None
+    kwargs["T"] = f'"{kwargs.get("T")}"' if kwargs.get("T") else None
+
+    with Session() as lib:
+        try:
+            arg_str = " ".join(["begin", f"{nrows}x{ncols}", build_arg_string(kwargs)])
+            lib.call_module("subplot", arg_str)
+            yield
+        finally:
+            v_arg = build_arg_string({"V": kwargs.get("V")})
+            lib.call_module("subplot", f"end {v_arg}")
+
+
+@fmt_docstring
+@contextlib.contextmanager
+@use_alias(A="fixedlabel", C="clearance", V="verbose")
+def set_panel(self, panel=None, **kwargs):
+    r"""
+    Set the current subplot panel to plot on.
+
+    Before you start plotting you must first select the active subplot. Note:
+    If any *projection* option is passed with **?** as scale or width when
+    plotting subplots, then the dimensions of the map are automatically
+    determined by the subplot size and your region. For Cartesian plots: If you
+    want the scale to apply equally to both dimensions then you must specify
+    ``projection="x"`` [The default ``projection="X"`` will fill the subplot by
+    using unequal scales].
+
+    {aliases}
+
+    Parameters
+    ----------
+    panel : str or list
+        *row,col*\|\ *index*.
+        Sets the current subplot until further notice. **Note**: First *row*
+        or *col* is 0, not 1. If not given we go to the next subplot by order
+        specified via **autolabel** in :meth:`pygmt.Figure.subplot`. As an
+        alternative, you may bypass the **sca** mode and instead supply the
+        common option **panel**=\ [*row,col*] to the first plot command you
+        issue in that subplot. GMT maintains information about the current
+        figure and subplot. Also, you may give the one-dimensional *index*
+        instead which starts at 0 and follows the row or column order set via
+        **autolabel** in :meth:`pygmt.Figure.subplot`.
```

Comment:
Incorrectly rendered, and `sca` no longer makes sense.

Reply:
Ok, fixed rendering using `**panel**\ =[*row,col*]`, and swapped `sca` to `:meth:pygmt.Figure.set_panel`